### PR TITLE
Push fixes

### PR DESCRIPTION
**Pull request by tomczoink**
Fixes and improvements to push docs per #679.

### CODE REVIEW

File: content/general/push/activate-subscribe.textile [hunk 1]
```diff
@@ -449,4 +430,19 @@ h3(#push-channel).
     default: PushChannel
     ruby:    Ably::Models::PushChannel
 
-<%= partial partial_version('types/_push_channel') %>
\ No newline at end of file
+<%= partial partial_version('types/_push_channel') %>
+
+h3(#push-channel-subscription).
+  default: PushChannelSubscription
+  ruby:    Ably::Models::PushChannelSubscription
```
Comment (mattheworiordan) [inline content/general/push/activate-subscribe.textile]: Is this right i.e. only Ruby has a different type?

File: content/rest/push.textile [hunk 2]
```diff
@@ -11,4 +11,80 @@ jump_to:
...
+      - removeWhere(params)#device-remove-where
+    - PushChannelSubscriptions:
+      - get(PushChannelSubscription)#push-channel-sub-get
+      - list(params)#push-channel-sub-list
+      - listChannels(params)#push-channel-sub-list-channels
+      - save(PushChannelSubscription)#push-channel-sub-save
+      - remove(PushChannelSubscription)#push-channel-sub-remove
+      - removeWhere(params)#push-channel-sub-remove-where
+  Push Device:
+    - Methods:
```
Comment (mattheworiordan) [inline content/rest/push.textile]: These don't exist in the REST API.  The REST API only offers admin functionality.

File: content/rest/push.textile [hunk 3]
```diff
@@ -11,4 +11,80 @@ jump_to:
...
+      - activate()#activate
+      - deactivate()#deactivate
+  Related types:
+    - DeviceDetails#device-details
+    - PushChannelSubscription#push-channel-subscription
+    - LocalDevice#local-device
```
Comment (mattheworiordan) [inline content/rest/push.textile]: I don't thikn this object exists either in REST API, worht checking though

File: content/rest/push.textile [hunk 4]
```diff
@@ -11,4 +11,80 @@ jump_to:
...
+    - PushChannel#push-channel
+    - PaginatedResult#paginated-result
+
+<%= partial partial_version('types/_push_admin') %>
+
+<div lang="android,swift,objc">
+
+<%= partial partial_version('types/_push_device') %>
```
Comment (mattheworiordan) [inline content/rest/push.textile]: I don't think this should be included in the REST API docs, this is only available in the Realtime API

**Comment by tomczoink** [comment]
Changes made per feedback @mattheworiordan, @paddybyers when you get a chance to give this a look over hopefully this should be good to merge.

**Pull request merged**